rrs/tools/rrs_upstream_email.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Send email to maintainers about the current status of the recipes.
+#
+# Copyright (C) 2015, 2018 Intel Corporation
+# Author: Aníbal Limón <anibal.limon@linux.intel.com>
+#
+# Licensed under the MIT license, see COPYING.MIT for details
+#
+# SPDX-License-Identifier: MIT
+
+import sys
+import os.path
+import optparse
+import logging
+from collections import namedtuple
+
+sys.path.insert(0, os.path.realpath(os.path.join(os.path.dirname(__file__))))
+from common import common_setup, get_pv_type, get_logger
+common_setup()
+from layerindex import utils
+
+utils.setup_django()
+from django.core.mail import EmailMessage
+from django.template.loader import get_template
+from django.template import Context, Template
+from django.core.urlresolvers import reverse
+from django.contrib.sites.models import Site
+import settings
+
+from layerindex.models import LayerItem, LayerBranch, Recipe
+from rrs.models import Maintainer, RecipeMaintainerHistory, RecipeMaintainer, \
+    RecipeUpstream, RecipeUpstreamHistory, MaintenancePlan
+
+logger = get_logger('UpstreamEmail', settings)
+
+
+"""
+    Send email with Recipes that need update.
+"""
+def send_email(maintplan, recipes, options):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 43 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 98% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code blocks on lines 69-77, lines 80-121 into separate functions.